dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/DiscoveryRestRepository.java

```diff
@@ -102,7 +102,13 @@ public SearchResultsRest getSearchObjects(final String query, final String dsoTy
         try {
             discoverQuery = queryBuilder
                 .buildQuery(context, scopeObject, discoveryConfiguration, query, searchFilters, dsoType, page);
-            searchResult = searchService.search(context, scopeObject, discoverQuery);
+
+            if (configuration != null && configuration.contentEquals("showUnDiscoverableItems")) {
```

Comment:
I'd prefer to minimize the amount of "exceptions" where a different method is used based on the name of the configuration because it creates code which is hard to maintain or debug.
It would be much cleaner if this is only based on the `defaultFilterQueries` in discovery.xml. But that on the other hand has a risk of being forgotten
If there would be a need for multiple configurations containing `UnDiscoverableItems` (e.g. only withdrawn but not filtering on discoverable, or different facets, or type-based), this method would need to be modified again.

Having a separate service method for including withdrawn and private items is actually not specific to this PR, but was already present in the service

Reply:
I agree.  I added this when I saw the `includeUnDiscoverable` boolean available in the public method parameters. 

I wasn't convinced that this step was actually needed and agree that relying on `defaultFilterQueries` in discovery.xml is probably better. 

As someone who has limited experience with solr it's disconcerting to see a query that excludes withdrawn items and then explicitly includes them at a later point. But that magic makes for easier to maintain code at this level 

`q=*:*&fl=dateIssued.year,handle,search.resourcetype,search.resourceid&fq=NOT(withdrawn:true)&fq=NOT(discoverable:false)&fq=search.resourcetype:2&fq=withdrawn:true+OR+discoverable:false&fq=location:me1ba3b9d-d403-4e57-a0d3-a08c782fe427&fq=dateIssued.year:[*+TO+*]&fq=read:(gadf81151-f3b6-4b29-9c09-6f28865fb614+OR+gadf81151-f3b6-4b29-9c09-6f28865fb614)&start=0&rows=1&sort=dateIssued.year_sort+asc`

Reply:
This query was not a valid example. 

As it turns out, the problem is solved by removing the boolean from the search methods. I didn't want to do this at first. But when I discovered that it's used only by the browse system (Jira issue indicates that support for withdrawn/private items needs to be removed from the browse system) I decided this change was in scope.